Public/FindEmptySpot.ps1

```diff
@@ -13,17 +13,17 @@
     Returns a tuple that contains the row and column of the empty spot. If there is no empty spot available, returns $null
 #>
 Function FindEmptySpot {
-    param(
+    Param(
         [parameter(Mandatory=$true)]
         [System.Object]$SudokuGrid
     )
     
-    for ($i = 0; $i -lt 9; $i++) {
-        for($j = 0; $j -lt 9; $j++) {
-            if ($SudokuGrid[$i][$j] -eq '-') {
-                return [System.Tuple]::Create(($i+1),($j+1))
+    For ($i = 0; $i -lt 9; $i++) {
+        For($j = 0; $j -lt 9; $j++) {
+            If ($SudokuGrid[$i][$j] -notmatch '\d') {
```

Comment:
If the goal of this PR was to add support for strings as well as integers, wouldn't this cause problems?

Reply:
Your original code was:
If ($SudokuGrid[$i][$j] -eq "-")

That relies on the empty spots to be "-".

The code I used:
If ($SudokuGrid[$i][$j] -notmatch '\d')

This relies on the empty spots to Not match any number.
So it shouldn't cause any issues and it will future proof that bit of code if you ever decide to use a different character for empty spots.

If you wanted it to be a little more dialed in (lets say you went back to using 0s for empty spots) you could use this instead:
If ($SudokuGrid[$i][$j] -match '[^1-9]')

This relies on the empty spots to match "Not 1 through 9".
So using either regex matching I provided will future proof that part of the code and shouldn't cause any issues.
It worked properly in my limited testing, but I probably don't have as robust testing setup as you do.